src/osx/cocoa/dataview.mm

```diff
@@ -1723,20 +1733,15 @@ -(void) actionDoubleClick:(id)sender
 // Default enter key behaviour is to begin cell editing. Subclass keyDown to
 // provide a keyboard wxEVT_DATAVIEW_ITEM_ACTIVATED event and allow the NSEvent
 // to pass if the wxEvent is not processed.
-- (void)keyDown:(NSEvent *)event
+
+// catch events routed here and feed them back before things get routed up the responder chain
+// otherwise we loose functionality like arrow keys etc.
+- (void)doCommandBySelector:(SEL)aSelector
 {
-    if( [[event charactersIgnoringModifiers]
-         characterAtIndex: 0] == NSCarriageReturnCharacter )
+    wxWidgetCocoaImpl* impl = (wxWidgetCocoaImpl* ) wxWidgetImpl::FindFromWXWidget( self );
+    if (impl)
     {
-        wxDataViewCtrl* const dvc = implementation->GetDataViewCtrl();
-        const wxDataViewItem item = wxDataViewItem( [[self itemAtRow:[self selectedRow]] pointer]);
-        wxDataViewEvent eventDV(wxEVT_DATAVIEW_ITEM_ACTIVATED, dvc, item);
-        if ( !dvc->GetEventHandler()->ProcessEvent(eventDV) )
-            [super keyDown:event];
-    }
-    else
-    {
-        [super keyDown:event];  // all other keys
+        impl->doCommandBySelector(aSelector, self, _cmd);
```

Comment:
I wonder if we could just remove this and add `doCommandBySelector:` to `wxOSXCocoaClassAddWXMethods()`? I.e. we currently do this for `wxGLCanvas`, more or less for `wxNonOwnedWindow` (it uses `shouldHandleSelector()` in addition, but this one could be made virtual in `wxWidgetCocoaImpl` if necessary), and `wxNSView((TextInput)`. Would there be any harm in doing this for all windows?

Of course, this doesn't have to be done in this PR, it's just a question for the future.

Reply:
I'm very hesitant to do so, already here in dataview things were not as I thought they'd be (see below)